### PR TITLE
Clothing equip and strip delays

### PR DESCRIPTION
**Pull request by TsunamiAnt**
## About The Pull Request
I was going to make this as a joke, but when I brought it up, quite a few people actually were interested in it. So here it is:
This adds an around 1 second delay to putting on most clothes, smaller or bigger depending on clothes of course. Hardsuits take longer, hats/glasses take less time/none at all.

Honestly in normal crew gameplay this is inconsequential. But during chases? When disguising yourself? Ooh boy.

## Why It's Good For The Game
Like always, delays are universally positive and deepen any system they touch. Your hardsuit slowing you down? Well, what's faster? Taking it off and putting it back on(my immersion!!)? Or just keeping it on? Now it's an actual choice! Consequences! Meaningful choices! Weight to our decision making!

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/153900881/88fc5aac-00f0-47b6-8654-d191fcfb235c

</details>

## Changelog
:cl:
add: Added Clothing equip delay and bubble announce
add: Added Clothing strip delay and bubble announce
/:cl:

### CODE REVIEW

**Comment by MarkusLarsson421** [comment]
Unsure how I feel about this. Disguises are already hard enough given the heavy inventory restriction.

**Comment by TsunamiAnt** [comment]
changing which gear gets delays and which doesn't is very easy with this, so I can definitely do some tweaking there yeah.

also yup it's a do_after, so no moving till you're done.

**Comment by JixS4v** [comment]
Maybe make it so Chameleon stuff takes less time to put on? It's meant to be a quick disguise tool so it would fit, probably

**Comment by TsunamiAnt** [comment]
> Maybe make it so Chameleon stuff takes less time to put on?

I like this actually, and we can get as specific as we want with the timers. If anyone has more suggestions like this keep em coming! I'll be putting all those in over time(heh).

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 1]
```diff
@@ -155,16 +155,23 @@
...
 	if(index && !QDELETED(src) && dna.species.mutanthands) //hand freed, fill with claws, skip if we're getting deleted.
 		put_in_hand(new dna.species.mutanthands(), index)
+	if(!bypass_delay)
+		if(!(I in held_items) && I == head || I == wear_suit || I == shoes || I == gloves || I == wear_mask || I == w_uniform) //only apply to shit in said slots, never to items that aren't actually worn.
+			if (I.strip_delay_self)
+				var/mob/living/carbon/human/H = src
+				H.visible_message("<span class='notice'>[H] starts taking off [I]...</span>", "<span class='notice'>You start taking off [I]...</span>")
+				H.balloon_alert_to_viewers("[H] starts taking off [I]...", "You start taking off [I]...")
+				playsound(H, 'sound/effects/equip.ogg', 60)
+				if(!do_after(H, I.strip_delay_self))
```
Comment (XeonMations) [inline code/modules/mob/living/carbon/human/inventory.dm:174]: > also yup it's a do_after, so no moving till you're done.

do_after's actually have a few variables which you can designate that alter its behavior, like if the timer is reset when the target is moving or not, for example.

**Comment by MarkusLarsson421** [comment]
> Maybe make it so Chameleon stuff takes less time to put on? It's meant to be a quick disguise tool so it would fit, probably 

Unfortunately for the chameleon gear, that's something you would put on and more or less never take off unless you're putting on a hardsuit.

**Comment by MarkusLarsson421** [comment]
How will this affect job specific bags? Line the chem bag, construction bag, ore bag and so on.

File: code/modules/mob/living/carbon/human/inventory.dm (hunk 1 above)
Comment (TsunamiAnt) [reply code/modules/mob/living/carbon/human/inventory.dm:174]: damn, didn't know that. But having to stand still is better.

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 2]
```diff
@@ -236,6 +243,9 @@
 		s_store = null
 		if(!QDELETED(src))
 			update_inv_s_store()
+	. = ..(I, force, newloc, no_move, invdrop, was_thrown) //See mob.dm for an explanation on this and some rage about people copypasting instead of calling ..() like they should.
+	if(!. || !I)
+		return
```
Comment (XeonMations) [inline code/modules/mob/living/carbon/human/inventory.dm]: By the way, you can fix your runtime error by moving this to the top of the proc call :)

The runtime you're having is an issue of I being null, and thus passing the I ==wear_suit check at the top of the proc at line 172, so by moving this up it exits early due to that.

File: code/modules/mob/living/carbon/human/inventory.dm (hunk 1 above)
Comment (MarkusLarsson421) [reply code/modules/mob/living/carbon/human/inventory.dm:174]: Depends on the item in question.

**Pull request closed without merging**